opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/sitecustomize.py

```diff
@@ -12,124 +12,54 @@
+    entry_point.load()()
 
-    for entry_point in iter_entry_points("opentelemetry_instrumentor"):
-        if entry_point.name in package_to_exclude:
-            logger.debug(
-                "Instrumentation skipped for library %s", entry_point.name
-            )
-            continue
+try:
+    disabled_instrumentations = set(
+        environ.get(OTEL_PYTHON_DISABLED_INSTRUMENTATIONS, "").split()
+    )
 
-        try:
-            conflict = get_dist_dependency_conflicts(entry_point.dist)
-            if conflict:
-                logger.debug(
-                    "Skipping instrumentation %s: %s",
-                    entry_point.name,
-                    conflict,
-                )
-                continue
+    for entry_point in iter_entry_points("opentelemetry-instrumentor"):
+        if entry_point.name in disabled_instrumentations:
+            _logger.debug("Instrumentation skipped for %s", entry_point.name)
 
-            # tell instrumentation to not run dep checks again as we already did it above
-            distro.load_instrumentor(entry_point, skip_dep_check=True)
-            logger.debug("Instrumented %s", entry_point.name)
-        except Exception as exc:  # pylint: disable=broad-except
-            logger.exception("Instrumenting of %s failed", entry_point.name)
-            raise exc
+        conflicts = get_dist_dependency_conflicts(entry_point.dist)
 
+        if conflicts is not None:
 
-def _load_configurators():
-    configured = None
-    for entry_point in iter_entry_points("opentelemetry_configurator"):
-        if configured is not None:
-            logger.warning(
-                "Configuration of %s not loaded, %s already loaded",
+            _logger.debug(
+                "Skipping instrumentation %s: %s",
                 entry_point.name,
-                configured,
+                conflicts,
             )
-            continue
-        try:
-            entry_point.load()().configure()  # type: ignore
-            configured = entry_point.name
-        except Exception as exc:  # pylint: disable=broad-except
-            logger.exception("Configuration of %s failed", entry_point.name)
-            raise exc
-
-
-def initialize():
-    try:
-        distro = _load_distros()
-        distro.configure()
-        _load_configurators()
-        _load_instrumentors(distro)
-    except Exception:  # pylint: disable=broad-except
-        logger.exception("Failed to auto initialize opentelemetry")
-    finally:
-        environ["PYTHONPATH"] = sub(
-            r"{}{}?".format(dirname(abspath(__file__)), pathsep),
-            "",
-            environ["PYTHONPATH"],
-        )
-
-
-if (
-    hasattr(sys, "argv")
-    and sys.argv[0].split(path.sep)[-1] == "celery"
-    and "worker" in sys.argv[1:]
-):
-    from celery.signals import worker_process_init  # pylint:disable=E0401
 
-    @worker_process_init.connect(weak=False)
-    def init_celery(*args, **kwargs):
```

Comment:
I understand we want to move this special case out of here but is this the right PR to do it? May be once we have the whole picture sorted out we can move it then to somewhere better suited for this. I'm concerned if we remove this now without having another proper place for it, we might forget about it and cause regressions in the next release.

Reply:
Yes, I mistakenly deleted it. I guess we can find a solution for Celery that uses the pre-instrument entry point.

Reply:
Yes, that'd probably be a good place for it and the hook can live in celery instrumentation package. Makes sense to me.